sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -0,0 +1,361 @@
+                              &platVer.front(), NULL);
+
+  if (ret_err != CL_SUCCESS) {
+    *queue = nullptr;
+    return PI::cast<pi_result>(ret_err);
+  }
+
+  if (platVer.find("OpenCL 1.0") != std::string::npos ||
+      platVer.find("OpenCL 1.1") != std::string::npos ||
+      platVer.find("OpenCL 1.2") != std::string::npos) {
+    *queue = PI::cast<pi_queue>(clCreateCommandQueue(
+        PI::cast<cl_context>(context), PI::cast<cl_device_id>(device),
+        PI::cast<cl_command_queue_properties>(properties), &ret_err));
+    return PI::cast<pi_result>(ret_err);
+  }
+
+  cl_queue_properties CreationFlagProperties[] = {
+      CL_QUEUE_PROPERTIES, PI::cast<cl_command_queue_properties>(properties),
+      0};
+  *queue = PI::cast<pi_queue>(clCreateCommandQueueWithProperties(
+      PI::cast<cl_context>(context), PI::cast<cl_device_id>(device),
+      CreationFlagProperties, &ret_err));
+  return PI::cast<pi_result>(ret_err);
+}
+
+pi_result OCL(piProgramCreate)(pi_context context, const void *il,
+                               size_t length, pi_program *res_program) {
+
+  size_t deviceCount;
+
+  cl_int ret_err = clGetContextInfo(PI::cast<cl_context>(context),
+                                    CL_CONTEXT_DEVICES, 0, NULL, &deviceCount);
+
+  std::vector<cl_device_id> devicesInCtx(deviceCount);
+
+  ret_err = clGetContextInfo(PI::cast<cl_context>(context), CL_CONTEXT_DEVICES,
+                             deviceCount * sizeof(cl_device_id),
+                             devicesInCtx.data(), NULL);
+
+  if (ret_err != CL_SUCCESS || deviceCount < 1) {
+    if (res_program != nullptr)
+      *res_program = nullptr;
+    return PI::cast<pi_result>(CL_INVALID_CONTEXT);
+  }
+
+  cl_platform_id curPlatform;
+  ret_err = clGetDeviceInfo(devicesInCtx[0], CL_DEVICE_PLATFORM,
+                            sizeof(cl_platform_id), &curPlatform, NULL);
+
+  if (ret_err != CL_SUCCESS) {
+    if (res_program != nullptr)
+      *res_program = nullptr;
+    return PI::cast<pi_result>(CL_INVALID_CONTEXT);
+  }
+
+  size_t devVerSize;
+  ret_err =
+      clGetPlatformInfo(curPlatform, CL_PLATFORM_VERSION, 0, NULL, &devVerSize);
+  std::string devVer(devVerSize, '\0');
+  ret_err = clGetPlatformInfo(curPlatform, CL_PLATFORM_VERSION, devVerSize,
+                              &devVer.front(), NULL);
+
+  if (ret_err != CL_SUCCESS) {
+    if (res_program != nullptr)
+      *res_program = nullptr;
+    return PI::cast<pi_result>(CL_INVALID_CONTEXT);
+  }
+
+  pi_result err = PI_SUCCESS;
+  if (devVer.find("OpenCL 1.0") == std::string::npos &&
+      devVer.find("OpenCL 1.1") == std::string::npos &&
+      devVer.find("OpenCL 1.2") == std::string::npos &&
+      devVer.find("OpenCL 2.0") == std::string::npos) {
+    if (res_program != nullptr)
+      *res_program = PI::cast<pi_program>(clCreateProgramWithIL(
+          PI::cast<cl_context>(context), il, length, PI::cast<cl_int *>(&err)));
+    return err;
```

Comment:
If res_program is nullptr, I think this PI call should return PI_INVALID_VALUE but currently it will return PI_SUCCESS

Reply:
done.